src/bsoncxx/types/bson_value/value.cpp

```diff
@@ -150,9 +157,9 @@ value::value(stdx::string_view code, bsoncxx::document::view_or_value scope)
     _impl->_value.value.v_codewscope.code = make_copy_for_libbson(code);
     _impl->_value.value.v_codewscope.code_len = (uint32_t)code.length();
     _impl->_value.value.v_codewscope.scope_len = (uint32_t)scope.view().length();
-    _impl->_value.value.v_codewscope.scope_data = (uint8_t*)bson_malloc0(scope.view().length());
-    std::memcpy(
-        _impl->_value.value.v_codewscope.scope_data, scope.view().data(), scope.view().length());
+
+    bson_t* cpy = bson_new_from_data(scope.view().data(), scope.view().length());
+    _impl->_value.value.v_codewscope.scope_data = bson_destroy_with_steal(cpy, true, &cpy->len);
```

Comment:
Sorry, when I initially suggested `bson_new_from_data` I forgot that `scope_data` is storing a pointer to the data directly, not a `bson_t`. I had not realized you'd have to destroy the outer `bson_t`. This seems about the same in terms of readability. I'm fine with either. If you do want to change it back, I only suggest using `bson_malloc` instead of `bson_malloc0`.

Reply:
Done.